src/pages/ProfilesExplorerView/components/SceneLabelValuesTimeseries.tsx

```diff
@@ -118,22 +145,28 @@ export class SceneLabelValuesTimeseries extends SceneObjectBase<SceneLabelValues
     };
   }
 
-  buildMenuItems(scaleType = ScaleDistribution.Linear): PanelMenuItem[] {
-    return [
+  buildMenuItems({
+    scaleType,
+    link: addToInvestigationLink,
+  }: {
+    scaleType?: ScaleDistribution;
+    link?: PluginExtensionLink;
+  }): PanelMenuItem[] {
+    const existingItems = this.state.body.state.menu?.state.items;
+    let selectedScaleType = scaleType;
+
+    if (!selectedScaleType) {
+      // ugly but working
+      const selectedScaleIndex = existingItems?.[0].subMenu?.findIndex((i) => i.text[0] === '✔') as number;
```

Comment:
Suggestions welcome :)

Reply:
It's a good idea, so we can extract the menu logic to a new object. But I don't think we need a Scenes behaviour, just extending `VizPanelMenu` would be enough I guess.